src/coreclr/src/tools/ILVerification/Verifier.cs

```diff
@@ -183,6 +189,16 @@ private IEnumerable<VerificationResult> VerifyMethod(EcmaModule module, MethodIL
 
                 importer.ReportVerificationError = (args, code) =>
                 {
+                    switch (code)
+                    {
+                        case VerifierError.ThrowOrCatchOnlyExceptionType:
```

Comment:
I think it would better to not run the check in the first place. It would also make it clear in the verifier implementation that the given check is just a sanity check.

Reply:
Would it be better to create a separate delegate such as; `ReportSanityCheckViolation`? Then we can keep a single enable/disable check.

Reply:
For readability, I think it is important to see in `ILImporter.Verify.cs` code whether the check is verification check or sanity check. The easiest way to do that is to put the sanity checks under if condition.

Also, some of the sanity checks may end up being expensive and so running them just to throw away the results is not good.